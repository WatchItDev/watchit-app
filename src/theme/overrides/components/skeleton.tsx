import { Theme } from '@mui/material/styles';

// ----------------------------------------------------------------------

export function skeleton(theme: Theme) {
  return {
    MuiSkeleton: {
      styleOverrides: {
        root: {
          backgroundColor: theme.palette.background.neutral,
        },
        rounded: {
          borderRadius: theme.shape.borderRadius * 2,
        },
      },
    },
  };
}
