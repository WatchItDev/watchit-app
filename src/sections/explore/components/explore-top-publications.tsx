import CarouselTopTitles from '@src/components/carousel/variants/carousel-top-titles.tsx';
import { useGetRecentPostsQuery } from '@src/graphql/generated/hooks.tsx';
import { ExploreTopPublicationsSkeleton } from '@src/sections/explore/components/explore-top-publications.skeleton.tsx';
import { LoadingFade } from '@src/components/LoadingFade.tsx';

// ----------------------------------------------------------------------

export const ExploreTopPublications = () => {
  const { data: allData, loading } = useGetRecentPostsQuery({
    variables: { limit: 10 },
  });

  return (
    <LoadingFade
      loading={loading}
      skeleton={<ExploreTopPublicationsSkeleton />}
    >
      <CarouselTopTitles posts={allData?.getRecentPosts ?? []} />
    </LoadingFade>
  );
};
