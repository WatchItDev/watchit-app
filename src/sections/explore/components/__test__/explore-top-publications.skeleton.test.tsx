import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import { ExploreTopPublicationsSkeleton } from "../explore-top-publications.skeleton";

describe("Testing in the <explore-top-publications.skeleton/> component", () => {
  it("should match snapshot", () => {
    const { container } = render(<ExploreTopPublicationsSkeleton />);
    expect(container).toMatchSnapshot();
  });
});
