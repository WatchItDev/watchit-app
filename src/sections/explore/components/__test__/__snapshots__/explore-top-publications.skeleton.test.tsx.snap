// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing in the <explore-top-publications.skeleton/> component > should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1lqjy3x"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-18q1832-MuiSkeleton-root"
      style="width: 100%; height: 100%;"
    />
    <div
      class="MuiBox-root css-qd97tw"
    >
      <div
        class="MuiBox-root css-fbjsum"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-w33rqg-MuiSkeleton-root"
        />
        <div
          class="MuiBox-root css-i54c5d"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-13s1xpp-MuiSkeleton-root"
            style="width: 14rem; height: 2.5rem;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-13s1xpp-MuiSkeleton-root"
            style="width: 8rem; height: 2rem;"
          />
          <div
            class="MuiBox-root css-8dboi"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-omo2pj-MuiSkeleton-root"
              style="height: 1.2rem;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-omo2pj-MuiSkeleton-root"
              style="height: 1.2rem;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-omo2pj-MuiSkeleton-root"
              style="height: 1.2rem;"
            />
          </div>
          <div
            class="MuiBox-root css-tsefwf"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-13s1xpp-MuiSkeleton-root"
              style="width: 8rem; height: 2.5rem;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-13s1xpp-MuiSkeleton-root"
              style="width: 2.5rem; height: 2.5rem;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
